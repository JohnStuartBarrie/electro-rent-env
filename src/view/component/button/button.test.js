import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './index.js';

test('render a Button', () => {
  const wrapper = mount(
      <Button>My Button</Button>
  );
  expect(wrapper).toMatchSnapshot();
});

test('click on the Button', () => {
  const onclick = jest.fn();
  const wrapper = shallow(
      <Button onClick={onclick}>My Button</Button>
  );
  wrapper.simulate('click')
  expect(onclick).toHaveBeenCalled()
});
