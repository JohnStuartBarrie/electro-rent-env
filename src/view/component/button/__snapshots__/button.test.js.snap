// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Button 1`] = `
<Button>
  <button>
    My Button
  </button>
</Button>
`;
